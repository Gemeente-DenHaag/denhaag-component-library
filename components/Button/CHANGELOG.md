# @gemeente-denhaag/button

## 0.2.3

### Patch Changes

- Updated dependencies [281fa34]
  - @gemeente-denhaag/icons@1.0.0
