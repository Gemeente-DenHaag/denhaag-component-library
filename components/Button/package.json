{
  "name": "@gemeente-denhaag/button",
  "description": "A Button component",
  "author": "Municipality of The Hague",
  "license": "EUPL-1.2",
  "version": "0.2.3-alpha.48",
  "main": "dist/index.js",
  "module": "dist/index.js",
  "exports": "./dist/index.js",
  "types": "dist/index.d.ts",
  "files": [
    "dist"
  ],
  "scripts": {
    "build": "rollup -c ../../rollup.config.js",
    "clean": "yarn rimraf dist tsconfig.tsbuildinfo"
  },
  "repository": {
    "type": "git",
    "url": "https://github.com/nl-design-system/denhaag.git",
    "directory": "components/Button"
  },
  "bugs": "https://github.com/nl-design-system/denhaag/issues",
  "dependencies": {
    "@gemeente-denhaag/baseprops": "^0.2.3-alpha.48",
    "@material-ui/core": "^4.11.0"
  },
  "devDependencies": {
    "@gemeente-denhaag/icons": "^0.2.3-alpha.48"
  },
  "peerDependencies": {
    "react": "^17.0.1"
  },
  "gitHead": "dcf72a9b79266c1ebede35aff4a02dd9121a980f"
}
