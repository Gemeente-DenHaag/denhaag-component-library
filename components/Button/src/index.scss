.denhaag-button {
  --denhaag-button-focus-border-offset: calc(-1 * var(--denhaag-focus-border-width));

  align-items: center;
  background-color: var(--denhaag-button-primary-action-background-color);
  border: 0;
  border-radius: var(--denhaag-button-border-radius);
  color: var(--denhaag-button-primary-action-color);
  cursor: var(--denhaag-button-cursor, default);
  display: inline-flex;
  font-family: var(--denhaag-button-font-family);
  font-weight: var(--denhaag-button-font-weight, 400);
  font-size: var(--denhaag-button-font-size, var(--denhaag-typography-scale-base-font-size));
  justify-content: center;
  line-height: 1.5;
  padding-block-end: var(--denhaag-button-medium-size-padding-block, var(--denhaag-button-padding-block));
  padding-block-start: var(--denhaag-button-medium-size-padding-block, var(--denhaag-button-padding-block));
  padding-inline-end: var(--denhaag-button-medium-size-padding-inline, var(--denhaag-button-padding-inline));
  padding-inline-start: var(--denhaag-button-medium-size-padding-inline, var(--denhaag-button-padding-inline));
  position: relative;
  text-decoration: none;
  transition-property: background-color, border-color, color, transform;
  transition-duration: var(--denhaag-button-transition-duration);
  transition-timing-function: var(--denhaag-button-transition-timing-function);
  transition-delay: var(--denhaag-button-transition-delay);
  transform: translate(var(--denhaag-button-transform-x, 0), var(--denhaag-button-transform-y, 0));

  @media (prefers-reduced-motion: reduce) {
    --denhaag-button-transition-duration: 0;
  }
}

.denhaag-button--reduced-button-transition {
  --denhaag-button-transition-duration: 0;
}

a.denhaag-button {
  --denhaag-button-cursor: pointer;
}

.denhaag-button.denhaag-button--hover,
.denhaag-button:hover {
  --denhaag-button-transform-y: var(--denhaag-button-hover-transform-y);

  background-color: var(--denhaag-button-primary-action-hover-background-color);
  color: var(--denhaag-button-primary-action-hover-color);
}

.denhaag-button--focus-visible::after,
.denhaag-button:focus-visible::after {
  border: var(--denhaag-focus-border);
  border-radius: var(--denhaag-border-radius);
  bottom: var(--denhaag-button-focus-border-offset);
  content: "";
  display: block;
  left: var(--denhaag-button-focus-border-offset);
  position: absolute;
  right: var(--denhaag-button-focus-border-offset);
  top: var(--denhaag-button-focus-border-offset);
}

.denhaag-button--focus,
.denhaag-button:focus,
.denhaag-button--focus-visible,
.denhaag-button:focus-visible {
  --denhaag-button-transform-y: var(--denhaag-button-focus-transform-y);

  outline: none;
}

.denhaag-button.denhaag-button--disabled,
.denhaag-button.denhaag-button:disabled {
  background-color: var(--denhaag-button-primary-action-disabled-background-color);
  color: var(--denhaag-button-primary-action-disabled-color);
}

.denhaag-button--secondary-action {
  // Due to the accessibility focus outline overlapping with the border, the outline needs to be repositioned for the border variant.
  --denhaag-button-focus-border-offset: calc(-1 * (var(--denhaag-focus-border-width) * 1.5));

  background-color: var(--denhaag-button-secondary-action-background-color);
  border-color: var(--denhaag-button-secondary-action-border-color);
  border-style: solid;
  border-width: var(--denhaag-button-border-width);
  color: var(--denhaag-button-secondary-action-color);
  padding-block-end: calc(
    var(--denhaag-button-medium-size-padding-block, var(--denhaag-button-padding-block)) -
      var(--denhaag-button-border-width)
  );
  padding-block-start: calc(
    var(--denhaag-button-medium-size-padding-block, var(--denhaag-button-padding-block)) -
      var(--denhaag-button-border-width)
  );
  padding-inline-end: calc(
    var(--denhaag-button-medium-size-padding-inline, var(--denhaag-button-padding-inline)) -
      var(--denhaag-button-border-width)
  );
  padding-inline-start: calc(
    var(--denhaag-button-medium-size-padding-inline, var(--denhaag-button-padding-inline)) -
      var(--denhaag-button-border-width)
  );
}

.denhaag-button--secondary-action.denhaag-button--hover,
.denhaag-button--secondary-action:hover {
  background-color: var(--denhaag-button-secondary-action-hover-background-color);
  border-color: var(--denhaag-button-secondary-action-hover-border-color);
  color: var(--denhaag-button-secondary-action-hover-color);
}

.denhaag-button--secondary-action.denhaag-button--disabled,
.denhaag-button--secondary-action.denhaag-button:disabled {
  background-color: var(--denhaag-button-secondary-action-disabled-background-color);
  border-color: var(--denhaag-button-secondary-action-disabled-border-color);
  color: var(--denhaag-button-secondary-action-disabled-color);
}

.denhaag-button--large {
  --denhaag-button-font-size: var(--denhaag-typography-scale-lg-font-size);
  --denhaag-button-medium-size-padding-block: var(--denhaag-button-large-size-padding-block);
  --denhaag-button-medium-size-padding-inline: var(--denhaag-button-large-size-padding-inline);

  line-height: var(--denhaag-button-large-size-line-height);
}

.denhaag-button--large.denhaag-button--secondary-action {
  --denhaag-button-medium-size-padding-block: calc(
    var(--denhaag-button-large-size-padding-block, var(--denhaag-button-padding-block)) -
      var(--denhaag-button-border-width)
  );
  --denhaag-button-medium-size-padding-inline: calc(
    var(--denhaag-button-large-size-padding-inline, var(--denhaag-button-padding-inline)) -
      var(--denhaag-button-border-width)
  );
}

.denhaag-button__icon > *:first-child {
  font-size: var(--denhaag-typography-scale-lg-font-size);
}

.denhaag-button--large .denhaag-button__icon > *:first-child {
  font-size: var(--denhaag-typography-scale-xl-font-size);
}

.denhaag-button--start-icon .denhaag-button__icon {
  display: inherit;
  margin-inline-end: 8px;
  margin-inline-start: -4px;
}

.denhaag-button--end-icon .denhaag-button__icon {
  display: inherit;
  margin-inline-end: -4px;
  margin-inline-start: 8px;
}

.denhaag-button__sr-only {
  position: absolute;
  width: 1px;
  height: 1px;
  padding-block-start: 0;
  padding-block-end: 0;
  padding-inline-start: 0;
  padding-inline-end: 0;
  margin-block-start: -1px;
  margin-block-end: -1px;
  margin-inline-start: -1px;
  margin-inline-end: -1px;
  overflow: hidden;
  clip: rect(0, 0, 0, 0);
  border: 0;
}

.denhaag-button--icon-only {
  --denhaag-button-medium-size-padding-inline: 0.75rem;

  height: var(--denhaag-button-icon-only-height);
}

.denhaag-button--icon-only .denhaag-button__icon {
  height: var(--denhaag-button-icon-only-icon-height);
}
