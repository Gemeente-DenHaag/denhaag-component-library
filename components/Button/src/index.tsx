import React from 'react';
import BaseProps from '@gemeente-denhaag/baseprops';
import './button.css';
import clsx from 'clsx';

export interface ButtonProps extends BaseProps {
  /**
   * Simple click handler
   */
  onClick?: (event: React.MouseEvent<HTMLButtonElement> | React.TouchEvent<HTMLButtonElement>) => void;

  /**
   * Size of the component
   */
  size?: 'medium' | 'large';

  /**
   * Disables Button
   */
  disabled?: boolean;

  /**
   * Button variant
   */
  variant?: 'primary-action' | 'secondary-action';

  /**
   * HTML type prop
   */
  type?: 'button' | 'submit' | 'reset';

  /**
   * Icon placed at the start of the button
   */
  startIcon?: React.ReactNode;

  /**
   * Icon placed at the end of the button
   */
  endIcon?: React.ReactNode;
}

/**
 * Primary UI component for user interaction
 */
export const Button: React.FC<ButtonProps> = (props: ButtonProps) => {
  const sizeClass = `denhaag-button--${props.size ?? 'medium'}`;
  const variantClass = `denhaag-button--${props.variant ?? 'primary-action'}`;
  const rootClassNames = clsx('denhaag-button', sizeClass, variantClass, props.className);

  const iconClassNames = clsx(
    {
      'denhaag-button__start-icon': props.startIcon !== undefined,
      'denhaag-button__end-icon': props.endIcon !== undefined,
    },
    `denhaag-button__icon-size-${props.size ?? 'medium'}`,
  );
  const iconWrapped = <span className={iconClassNames}>{props.startIcon || props.endIcon}</span>;

  return (
    <button className={rootClassNames} onClick={props.onClick} disabled={props.disabled} type={props.type}>
      {props.startIcon !== undefined ? iconWrapped : ''}
      {props.children}
      {props.endIcon !== undefined ? iconWrapped : ''}
    </button>
  );
};

/**
 * Default export for Button
 */
export default Button;
