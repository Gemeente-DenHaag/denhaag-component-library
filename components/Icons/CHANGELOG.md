# @gemeente-denhaag/icons

## 1.0.0

### Major Changes

- 281fa34: Added wallet icon
