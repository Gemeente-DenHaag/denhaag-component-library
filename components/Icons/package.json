{
  "name": "@gemeente-denhaag/icons",
  "version": "0.2.3-alpha.84",
  "description": "The icon set of the Municipality of The Hague",
  "bugs": "https://github.com/nl-design-system/denhaag/issues",
  "repository": {
    "type": "git",
    "url": "https://github.com/nl-design-system/denhaag.git",
    "directory": "components/Icons"
  },
  "license": "EUPL-1.2",
  "author": "Municipality of The Hague",
  "exports": "./dist/index.js",
  "main": "dist/index.js",
  "module": "dist/index.js",
  "types": "dist/index.d.ts",
  "files": [
    "dist"
  ],
  "scripts": {
    "build": "rollup -c ../../rollup.config.js",
    "clean": "yarn rimraf dist tsconfig.tsbuildinfo"
  },
  "dependencies": {
    "@gemeente-denhaag/baseprops": "0.2.3-alpha.84",
    "@material-ui/core": "4.12.3"
  },
  "peerDependencies": {
    "react": "^17.0.0"
  }
}
