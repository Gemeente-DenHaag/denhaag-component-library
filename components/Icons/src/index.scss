.denhaag-icon {
  aspect-ratio: 1;
  display: inline-block;
  fill: currentcolor;
  flex-shrink: 0;
  font-size: var(--denhaag-icon-size, var(--denhaag-space-xl));
  height: 1em;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  width: 1em;
}
