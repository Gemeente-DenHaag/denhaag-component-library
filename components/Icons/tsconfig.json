{
  "extends": "../../tsconfig.build.json",
  "compilerOptions": {
    "rootDir": "src",
    "outDir": "dist",
    "typeRoots": ["./src/@types"],
    "composite": true
  },
  "include": ["src/**/*"],
  "exclude": ["dist", "src/**/*.stories.tsx"]
}
