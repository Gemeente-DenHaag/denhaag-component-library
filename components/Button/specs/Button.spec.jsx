import * as React from 'react';
import Button from '../src';
import { ChevronRightIcon } from '../../Icons';

describe('Button tests', () => {
  // Starting with functionality tests
  it('can be clicked', () => {
    const props = {
      onClick: () => {
        console.log('Clicked');
      },
    };

    const spy = cy.spy(props, 'onClick').as('Button click');

    cy.mount(<Button onClick={props.onClick}>Test button</Button>);

    cy.get('.denhaag-button')
      .contains('Test button')
      .click()
      .then(() => expect(spy).to.have.been.calledOnce);
  });

  it('cannot be clicked when disabled', () => {
    const props = {
      onClick: () => {
        console.log('Clicked');
      },
    };

    const spy = cy.spy(props, 'onClick').as('Button click');

    cy.mount(
      <Button onClick={props.onClick} disabled>
        Test button
      </Button>,
    );

    cy.get('.denhaag-button')
      .click({ force: true })
      .then(() => expect(spy).to.not.have.been.called);
  });

  it('can be operated with keyboard', () => {
    const props = {
      onClick: () => {
        console.log('Clicked');
      },
    };

    const spy = cy.spy(props, 'onClick').as('Button click');

    cy.mount(<Button onClick={props.onClick}>Test button</Button>);

    cy.get('.denhaag-button')
      .type('{enter}')
      .then(() => expect(spy).to.have.been.calledOnce);
  });

  it('a11y and snapshots', () => {
    cy.snapshots(
      <Button startIcon={<ChevronRightIcon />} endIcon={<ChevronRightIcon />}>
        Test button
      </Button>,
      {
        variant: 'secondary-action',
        size: 'large',
        disabled: true,
      },
      {
        selector: '.denhaag-button',
        states: {
          hover: 'denhaag-button--hover',
          focus: 'denhaag-button--focus',
        },
      },
    );
  });
});
