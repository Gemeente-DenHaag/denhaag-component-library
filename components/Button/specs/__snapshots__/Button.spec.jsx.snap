exports[`Button tests > a11y and snapshots #0`] = `
<div id="wrapper" style="display: inline-block;">
  <div style="padding: 10px;">
    <button type="button" class="denhaag-button">
      <span class="denhaag-button__label">Test button</span>
    </button>
  </div>
  <div style="padding: 10px;">
    <button
      variant="secondary-action"
      type="button"
      class="denhaag-button denhaag-button--secondary-action"
    >
      <span class="denhaag-button__label">Test button</span>
    </button>
  </div>
  <div style="padding: 10px;">
    <button disabled="" type="button" class="denhaag-button">
      <span class="denhaag-button__label">Test button</span>
    </button>
  </div>
  <div style="padding: 10px;">
    <button
      variant="secondary-action"
      type="button"
      class="denhaag-button denhaag-button--secondary-action denhaag-button--large"
    >
      <span class="denhaag-button__label">Test button</span>
    </button>
  </div>
</div>
`;

exports[`Button tests > a11y and snapshots #1`] = `
<div id="wrapper" style="display: inline-block;">
  <div style="padding: 10px;">
    <button type="button" class="denhaag-button denhaag-button--start-icon">
      <span class="denhaag-button__icon"
        ><svg
          xmlns="http://www.w3.org/2000/svg"
          viewBox="0 0 24 24"
          width="1em"
          height="1em"
          class="denhaag-icon"
          focusable="false"
          aria-hidden="true"
          shape-rendering="auto"
        >
          <path
            d="M19 13v4a2 2 0 01-2 2H3a2 2 0 01-2-2v-4M5 8l5 5 5-5M10 13V1"
            fill="none"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          ></path></svg></span
      ><span class="denhaag-button__label">Test button</span>
    </button>
  </div>
  <div style="padding: 10px;">
    <button
      variant="secondary-action"
      type="button"
      class="denhaag-button denhaag-button--secondary-action denhaag-button--start-icon"
    >
      <span class="denhaag-button__icon"
        ><svg
          xmlns="http://www.w3.org/2000/svg"
          viewBox="0 0 24 24"
          width="1em"
          height="1em"
          class="denhaag-icon"
          focusable="false"
          aria-hidden="true"
          shape-rendering="auto"
        >
          <path
            d="M19 13v4a2 2 0 01-2 2H3a2 2 0 01-2-2v-4M5 8l5 5 5-5M10 13V1"
            fill="none"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          ></path></svg></span
      ><span class="denhaag-button__label">Test button</span>
    </button>
  </div>
  <div style="padding: 10px;">
    <button type="button" class="denhaag-button denhaag-button--end-icon">
      <span class="denhaag-button__label">Test button</span
      ><span class="denhaag-button__icon"
        ><svg
          xmlns="http://www.w3.org/2000/svg"
          viewBox="0 0 24 24"
          width="1em"
          height="1em"
          class="denhaag-icon"
          focusable="false"
          aria-hidden="true"
          shape-rendering="auto"
        >
          <path
            d="M19 13v4a2 2 0 01-2 2H3a2 2 0 01-2-2v-4M5 8l5 5 5-5M10 13V1"
            fill="none"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          ></path></svg
      ></span>
    </button>
  </div>
  <div style="padding: 10px;">
    <button
      variant="secondary-action"
      disabled=""
      type="button"
      class="denhaag-button denhaag-button--secondary-action denhaag-button--large denhaag-button--end-icon"
    >
      <span class="denhaag-button__label">Test button</span
      ><span class="denhaag-button__icon"
        ><svg
          xmlns="http://www.w3.org/2000/svg"
          viewBox="0 0 24 24"
          width="1em"
          height="1em"
          class="denhaag-icon"
          focusable="false"
          aria-hidden="true"
          shape-rendering="auto"
        >
          <path
            d="M19 13v4a2 2 0 01-2 2H3a2 2 0 01-2-2v-4M5 8l5 5 5-5M10 13V1"
            fill="none"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          ></path></svg
      ></span>
    </button>
  </div>
</div>
`;
